code/modules/client/preferences.dm

```diff
@@ -1550,6 +1553,27 @@ GLOBAL_LIST_INIT(bgstate_options, list(
 					if(new_pref_squad)
 						preferred_squad = new_pref_squad
 
+				if("prefnvg")
+					var/new_nvg_color = tgui_input_list(user, "Choose the color of your night-vision", "Character Preferences", GLOB.nvg_color_list)
+					if(!new_nvg_color)
+						return
+					night_vision_preference = new_nvg_color
+					switch(new_nvg_color)
+						if("Green")
+							night_vision_color = NV_COLOR_GREEN
+						if("White")
+							night_vision_color = NV_COLOR_WHITE
+						if("YELLOW")
+							night_vision_color = NV_COLOR_YELLOW
+						if("Orange")
+							night_vision_color = NV_COLOR_ORANGE
+						if("Red")
+							night_vision_color = NV_COLOR_RED
+						if("Blue")
+							night_vision_color = NV_COLOR_BLUE
+						else
+							night_vision_color = NV_COLOR_GREEN
```

Comment:
don't save this separate to the pref, just make an assoc list of "Green" = NV_COLOR_GREEN, etc, and look that up instead on liek 335 of helmetgarb.dm